packages/shared/lib/services/sync/sync.service.ts

```diff
@@ -252,13 +268,24 @@ export const verifyOwnership = async (nangoConnectionId: number, environment_id:
 };
 
 export const deleteSync = async (syncId: string): Promise<string> => {
-    await schema().from<Sync>(TABLE).where({ id: syncId }).del();
+    await schema().from<Sync>(TABLE).where({ id: syncId, deleted: false }).update({ deleted: true });
+
+    await syncOrchestrator.deleteSyncRelatedObjects(syncId);
 
     return syncId;
 };
 
-export const findSyncByConnections = async (connectionIds: number[]): Promise<Sync[]> => {
-    const results = await schema().select('*').from<Sync>(TABLE).whereIn('nango_connection_id', connectionIds);
+export const findSyncByConnections = async (connectionIds: number[], sync_name: string): Promise<Sync[]> => {
```

Comment:
This change should maybe resolve https://github.com/NangoHQ/nango/issues/753

and as mentioned in the community https://nango-community.slack.com/archives/C04ABR352H0/p1690270888189499

> I have removed my old sync and added new sync with new model but still my old sync is showing in activity and new sync is not running but it is deployed.
Which command I need to run for that